### ISSUE TITLE
Bug: ComposedLrScheduler serializes Cosine scheduler as Linear

### ISSUE THREAD

**Issue by Glimmering-Stallion** [tracel-ai/burn#4610]
Today, when trying to do some training runs, I found out in burn-optim's composed.rs, the `to_record` implementation for
`ComposedLrScheduler` incorrectly serializes the Cosine scheduler
as `LrSchedulerRecord::Linear`.

Current code:
`LrSchedulerItem::Cosine(item) => {
    LrSchedulerRecord::Linear(item.to_record::<B>())
}`

What I believe to be the expected behavior:
`LrSchedulerItem::Cosine(item) => {
    LrSchedulerRecord::Cosine(item.to_record::<B>())
}`

Impact:
When serializing/deserializing composed schedulers during training runs, Cosine schedulers
become incorrectly recorded as Linear schedulers, which then leads to incorrect LR restoration when loading training checkpoints and a final `_ => panic!("Invalid state")`.

Proposed fix:
Just replace `LrSchedulerRecord::Linear` with `LrSchedulerRecord::Cosine` in the Cosine match arm.

File:
`burn-optim/src/lr_scheduler/composed.rs`

**Comment by laggui** [comment]
Good catch 😬 Thanks for reporting!

**Comment by Glimmering-Stallion** [comment]
Awesome! Glad I could help 👍
